src/Raven.Client/Documents/TimeSeries/TimeSeriesOperations.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Documents.Operations.TimeSeries;
+using Raven.Client.Documents.Session.TimeSeries;
+using Sparrow;
+
+namespace Raven.Client.Documents.TimeSeries
+{
+    public class TimeSeriesOperations
+    {
+        private readonly IDocumentStore _store;
+        private readonly string _database;
+        private readonly MaintenanceOperationExecutor _executor;
+
+        public TimeSeriesOperations(IDocumentStore store)
+        {
+            _store = store;
+            _database = store.Database;
+            _executor = _store.Maintenance.ForDatabase(_database);
+        }
+
+        private TimeSeriesOperations(IDocumentStore store, string database)
+        {
+            _store = store;
+            _database = database;
+        }
+
+        /// <summary>
+        /// Register value names of a time-series
+        /// </summary>
+        /// <typeparam name="TCollection">Collection type</typeparam>
+        /// <typeparam name="TTimeSeriesEntry">Time-series type, must be derived from TimeSeriesEntry class</typeparam>
+        public Task Register<TCollection, TTimeSeriesEntry>() where TTimeSeriesEntry : TimeSeriesEntry
```

Comment:
all 'async' methods available in Client API should have 'Async' in the name e.g. RegisterAsync